lib/formats/odata.js

```diff
@@ -344,10 +350,17 @@ const rowStreamToOData = (fields, table, domain, originalUrl, query, inStream, t
       // per row, we do our asynchronous parsing, jam the result onto the
       // text resultstream, and call done to indicate that the row is processed.
       submissionToOData(fields, table, row, options).then((data) => {
+
+        const parentIdProperty = Object.keys(data[0] ?? true).filter(p => /^__.*-id$/.test(p))[0];
```

Comment:
I had no idea that `Object.keys()` could be called with `true`! Is `true` automatically converted to an object? I feel like a ternary might be a little clearer here, something like:

```js
const parentIdProperty = data.length !== 0
  ? Object.keys(data[0]).find(p => /^__.*-id$/.test(p))
  : null;
```

(Also using `find()`, which I think is a little more performant than `.filter()[0]`.)